recipes/boost/all/conanfile.py

```diff
@@ -855,103 +1001,194 @@ def _is_versioned_layout(self):
+        return {
+            "lzma": "xz_utils",
+            "iconv": "libiconv",
+            "python": None,  # FIXME: change to cpython when it becomes available
+        }.get(name, name)
+
     def package_info(self):
-        gen_libs = [] if self.options.header_only else tools.collect_libs(self)
+        self.cpp_info.filenames["cmake_find_package"] = "Boost"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "Boost"
+        self.cpp_info.names["cmake_find_package"] = "Boost"
+        self.cpp_info.names["cmake_find_package_multi"] = "Boost"
 
-        if self._is_versioned_layout:
-            version_tokens = str(self.version).split(".")
-            if len(version_tokens) >= 2:
-                major = version_tokens[0]
-                minor = version_tokens[1]
-                boost_version_tag = "boost-%s_%s" % (major, minor)
-                self.cpp_info.includedirs = [os.path.join(self.package_folder, "include", boost_version_tag)]
-
-        # List of lists, so if more than one matches the lib like serialization and wserialization
-        # both will be added to the list
-        ordered_libs = [[] for _ in range(len(lib_list))]
-
-        # The order is important, reorder following the lib_list order
-        missing_order_info = []
-        for real_lib_name in gen_libs:
-            for pos, alib in enumerate(lib_list):
-                if os.path.splitext(real_lib_name)[0].split("-")[0].endswith(alib):
-                    ordered_libs[pos].append(real_lib_name)
-                    break
-            else:
-                # self.output.info("Missing in order: %s" % real_lib_name)
-                if "_exec_monitor" not in real_lib_name:  # https://github.com/bincrafters/community/issues/94
-                    missing_order_info.append(real_lib_name)  # Assume they do not depend on other
+        self.cpp_info.components["diagnostic_definitions"].libs = []
+        self.cpp_info.components["diagnostic_definitions"].defines = ["BOOST_LIB_DIAGNOSTIC"]
+        self.cpp_info.components["diagnostic_definitions"].names["cmake_find_package"] = "diagnostic_definitions"
+        self.cpp_info.components["diagnostic_definitions"].names["cmake_find_package_multi"] = "diagnostic_definitions"
+
+        self.cpp_info.components["disable_autolinking"].libs = []
+        self.cpp_info.components["disable_autolinking"].defines = ["BOOST_ALL_NO_LIB"]
+        self.cpp_info.components["disable_autolinking"].names["cmake_find_package"] = "disable_autolinking"
+        self.cpp_info.components["disable_autolinking"].names["cmake_find_package_multi"] = "disable_autolinking"
+
+        self.cpp_info.components["dynamic_linking"].libs = []
+        self.cpp_info.components["dynamic_linking"].defines = ["BOOST_ALL_DYN_LINK"]
+        self.cpp_info.components["dynamic_linking"].names["cmake_find_package"] = "dynamic_linking"
+        self.cpp_info.components["dynamic_linking"].names["cmake_find_package_multi"] = "dynamic_linking"
```

Comment:
Review Martin's comment, I noticed these components but I cannot find where they are being used, any pointers?

Reply:
They are being defined by the "native" FindBoost cmake file (https://cmake.org/cmake/help/latest/module/FindBoost.html)

Reply:
Yes these are CMake's components.
Please see https://github.com/conan-io/conan-center-index/pull/2097#issuecomment-737517998 for concerns.